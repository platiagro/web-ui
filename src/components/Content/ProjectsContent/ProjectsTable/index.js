// CORE LIBS
import React, { useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';

// UI LIBS
import { SearchOutlined } from '@ant-design/icons';
import {
  Button,
  Input,
  Popconfirm,
  Skeleton,
  Space,
  Table,
  Tag,
  Typography,
} from 'antd';

const { Text } = Typography;

/**
 * Projects Table.
 * This component is responsible for displaying projects table.
 */
const ProjectsTable = ({
  loading,
  projects,
  selectedProjects,
  handleClickProject,
  handleClickDelete,
  handleFetchPaginatedProjects,
  handleSelectProjects,
}) => {
  const [searchText, setSearchText] = useState('');
  const intervalRef = useRef(null);
  const previousSearchText = useRef(null);
  const searchInputRef = useRef(null);

  useEffect(() => {
    if (searchText) {
      intervalRef.current = setTimeout(() => {
        previousSearchText.current = searchText;
        handleFetchPaginatedProjects(searchText);
      }, 1000);
    } else {
      if (previousSearchText.current) {
        intervalRef.current = setTimeout(() => {
          handleFetchPaginatedProjects();
        }, 1000);
      } else {
        clearTimeout(intervalRef.current);
      }
    }
    return () => clearTimeout(intervalRef.current);
  }, [searchText, handleFetchPaginatedProjects]);

  const handleSearch = (selectedKeys, confirm) => {
    confirm();
    setSearchText(selectedKeys[0]);
  };

  const handleReset = (clearFilters) => {
    clearFilters();
    setSearchText('');
  };

  const columnsConfig = [
    {
      title: <strong>Nome do Projeto</strong>,
      dataIndex: 'name',
      key: 'name',
      width: '20%',
      filterDropdown: ({
        setSelectedKeys,
        selectedKeys,
        confirm,
        clearFilters,
      }) => (
        <div style={{ padding: 8 }}>
          <Input
            ref={searchInputRef}
            placeholder={`Nome`}
            value={selectedKeys[0]}
            onChange={(e) =>
              setSelectedKeys(e.target.value ? [e.target.value] : [])
            }
            onPressEnter={() => handleSearch(selectedKeys, confirm)}
            style={{ width: 188, marginBottom: 8, display: 'block' }}
          />
          <Space>
            <Button
              type='primary'
              onClick={() => handleSearch(selectedKeys, confirm)}
              icon={<SearchOutlined />}
              size='small'
              style={{ width: 90 }}
            >
              Search
            </Button>
            <Button
              onClick={() => handleReset(clearFilters)}
              size='small'
              style={{ width: 90 }}
            >
              Reset
            </Button>
          </Space>
        </div>
      ),
      onFilterDropdownVisibleChange: (visible) => {
        if (visible) {
          setTimeout(() => searchInputRef.current.select());
        }
      },
      filterIcon: (filtered) => (
        <SearchOutlined style={{ color: filtered ? '#1890ff' : undefined }} />
      ),
      render: (value, record) =>
        loading ? (
          <Skeleton
            paragraph={{ rows: 1, width: 150 }}
            size='large'
            title={false}
          />
        ) : (
          <Button
            type='link'
            onClick={() => handleClickProject(record.uuid)}
            style={{ textAlign: 'left', width: '100%' }}
          >
            <Text ellipsis style={{ color: '#0050B3', width: '100%' }}>
              {value}
            </Text>
          </Button>
        ),
    },
    {
      title: <strong>Descrição</strong>,
      dataIndex: 'description',
      key: 'description',
      width: '30%',
      ellipsis: true,
      render: (description) =>
        loading ? (
          <Skeleton
            paragraph={{ rows: 1, width: 250 }}
            size='large'
            title={false}
          />
        ) : (
          <></>
        ),
    },
    {
      title: <strong>Tags</strong>,
      dataIndex: 'tags',
      key: 'tags',
      width: '28%',
      filters: [
        {
          text: 'Experimentação',
          value: 'Experimentação',
        },
        {
          text: 'Pré-implantação',
          value: 'Pré-implantação',
        },
        {
          text: 'Implantado',
          value: 'Implantado',
        },
      ],
      onFilter: (value, record) => {
        return record.tags ? record.tags.indexOf(value) === 0 : false;
      },
      render: (tags) => {
        if (loading) {
          return (
            <Skeleton
              paragraph={{ rows: 1, width: 200 }}
              size='large'
              title={false}
            />
          );
        } else {
          //TODO Remover mock depois que as tags forem implementadas
          tags = ['Experimentação', 'Pré-implantação', 'Implantado'];
          return (
            <>
              {tags.map((tag) => {
                if (tag === 'Experimentação') {
                  return (
                    <Tag color='purple' key='purpleTag'>
                      {tag}
                    </Tag>
                  );
                } else if (tag === 'Pré-implantação') {
                  return (
                    <Tag color='volcano' key='volcanoTag'>
                      {tag}
                    </Tag>
                  );
                } else {
                  return (
                    <Tag color='green' key='greenTag'>
                      {tag}
                    </Tag>
                  );
                }
              })}
            </>
          );
        }
      },
    },
    {
      title: <strong>Última modificação</strong>,
      dataIndex: 'updatedAt',
      key: 'updatedAt',
      width: '12%',
      sorter: (a, b) => {
        const dateA = new Date(a.updatedAt);
        const dateB = new Date(b.updatedAt);
        return dateA.getTime() - dateB.getTime();
      },
      sortDirections: ['descend', 'ascend'],
      render: (value) =>
        loading ? (
          <Skeleton
            paragraph={{ rows: 1, width: 70 }}
            size='large'
            title={false}
          />
        ) : (
          new Date(value).toLocaleString()
        ),
    },
    {
      title: <strong>Ação</strong>,
      dataIndex: 'action',
      key: 'action',
      width: '10%',
      render: (value, record) =>
        loading ? (
          <Skeleton
            paragraph={{ rows: 1, width: 70 }}
            size='large'
            title={false}
          />
        ) : (
          <Popconfirm
            title='Você tem certeza que deseja excluir esse projeto?'
            onConfirm={() => handleClickDelete(record.uuid)}
            okText='Sim'
            cancelText='Não'
          >
            <Button type='link'>Excluir</Button>
          </Popconfirm>
        ),
    },
  ];
  // RENDER
  return (
    <Table
      className='projectsTable'
      rowKey={(record) => record.uuid}
      rowSelection={{
        selectedRowKeys: selectedProjects,
        onChange: (selectedRowKeys, selectedRows) => {
          handleSelectProjects(selectedRowKeys);
        },
      }}
      dataSource={projects}
      columns={columnsConfig}
      pagination={false}
      //loading={loading}
      loading={Boolean(loading && projects.length === 0)}
    />
  );
};

// PROP TYPES
ProjectsTable.propTypes = {
  /** is loading */
  loading: PropTypes.bool.isRequired,
  /** projects list */
  projects: PropTypes.arrayOf(PropTypes.object).isRequired,
  /** selected projects list */
  selectedProjects: PropTypes.arrayOf(PropTypes.string).isRequired,
  /** projects table click project handle */
  handleClickProject: PropTypes.func.isRequired,
  /** projects table delete project handle */
  handleClickDelete: PropTypes.func.isRequired,
  /** fetch paginated projects handle */
  handleFetchPaginatedProjects: PropTypes.func.isRequired,
  /** projects table row selection project handle */
  handleSelectProjects: PropTypes.func.isRequired,
};

// EXPORT
export default ProjectsTable;
