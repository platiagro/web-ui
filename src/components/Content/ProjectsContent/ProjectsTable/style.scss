.ant-popover-inner-content {
  box-shadow: 0 0 10px;
}

.myProjectsTableContainer {
  padding: 20px;
  background-color: #fff;
}
