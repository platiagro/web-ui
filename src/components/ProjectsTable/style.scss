table {
  table-layout: fixed;
}

td {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  cursor: pointer;
}

.ant-table-body {
  height: 535px;
}
