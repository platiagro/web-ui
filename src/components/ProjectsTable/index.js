import React from 'react';

import PropTypes from 'prop-types';

import { Table, Tooltip } from 'antd';

import './style.scss';

const tableColumns = [
  {
    title: 'Nome do Projeto',
    dataIndex: 'projectName',
    key: 'projectName',
  },
  {
    title: 'Experimentos',
    dataIndex: 'experiments',
    key: 'experiments',
  },
  {
    title: 'Data de Criação',
    dataIndex: 'created',
    key: 'created',
  },
];

const ProjectsTable = ({ projectList, rowSelection }) => {
  projectList.forEach((project) => {
    const projectAux = project;
    projectAux.experiments = (
      <Tooltip title={project.experimentsList.join(', ')}>
        <span>{project.experimentsList.join(', ')}</span>
      </Tooltip>
    );
  });

  return (
    <Table
      onRow={(record, rowIndex) => {
        return {
          onDoubleClick: (event) => {
            event.preventDefault();
            // eslint-disable-next-line
            console.log(record, rowIndex);
          }, // double click row
        };
      }}
      rowSelection={rowSelection}
      dataSource={projectList}
      columns={tableColumns}
      pagination={{ pageSize: 9 }}
    />
  );
};

ProjectsTable.propTypes = {
  projectList: PropTypes.arrayOf(PropTypes.object).isRequired,
  rowSelection: PropTypes.shape({
    selectedRowKeys: PropTypes.array,
    onChange: PropTypes.func,
  }).isRequired,
};

export default ProjectsTable;
