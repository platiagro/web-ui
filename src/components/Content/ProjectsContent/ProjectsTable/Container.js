// CORE LIBS
import React, { useLayoutEffect } from 'react';
import { connect } from 'react-redux';
import { useHistory } from 'react-router-dom';

// COMPONENTS
import ProjectsTablePagination from '../ProjectsTablePagination/Container';
import ProjectsTable from './index';
import { MyProjectsEmptyPlaceholder } from 'components/EmptyPlaceholders';

// ACTIONS
import {
  fetchPaginatedProjects,
  selectProjects,
} from '../../../../store/projects/actions';
import { deleteProjectRequest } from '../../../../store/project/actions';

// ACTIONS
import { showNewProjectModal } from '../../../../store/ui/actions';

// DISPATCHS
const mapDispatchToProps = (dispatch) => {
  return {
    handleFetchPaginatedProjects: (name) =>
      dispatch(fetchPaginatedProjects(name, 1, 10)),
    handleDeleteProject: (projectUuid) =>
      dispatch(deleteProjectRequest(projectUuid)),
    handleShowNewProjectModal: (record) =>
      dispatch(showNewProjectModal(record)),
    handleSelectProjects: (record) => dispatch(selectProjects(record)),
  };
};

// STATES
const mapStateToProps = (state) => {
  return {
    loading: state.uiReducer.projectsTable.loading,
    projects: state.projectsReducer.projects,
    searchText: state.projectsReducer.searchText,
    selectedProjects: state.projectsReducer.selectedProjects,
  };
};

/**
 * Projects Table Container.
 * This component is responsible for create a logic container for projects table
 * with redux.
 *
 * @param props
 */
const ProjectsTableContainer = (props) => {
  // destructuring props
  const {
    loading,
    projects,
    searchText,
    selectedProjects,
    handleFetchPaginatedProjects,
    handleDeleteProject,
    handleShowNewProjectModal,
    handleSelectProjects,
  } = props;

  // CONSTANTS
  // getting history
  const history = useHistory();

  // HOOKS
  // did mount hook
  useLayoutEffect(() => {
    handleFetchPaginatedProjects();
  }, [handleFetchPaginatedProjects]);

  // HANDLERS
  // project click
  const handleClickProject = (projectUuid) =>
    history.push(`/projetos/${projectUuid}`);

  // RENDER
  return loading || searchText || (projects && projects.length > 0) ? (
    <div className='myProjectsTableContainer'>
      <ProjectsTable
        loading={loading}
        projects={projects}
        selectedProjects={selectedProjects}
        handleClickProject={handleClickProject}
        handleClickDelete={handleDeleteProject}
        handleFetchPaginatedProjects={handleFetchPaginatedProjects}
        handleShowNewProjectModal={handleShowNewProjectModal}
        handleSelectProjects={handleSelectProjects}
      />
      <br />
      <ProjectsTablePagination />
    </div>
  ) : (
    <MyProjectsEmptyPlaceholder />
  );
};

// EXPORT
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProjectsTableContainer);
