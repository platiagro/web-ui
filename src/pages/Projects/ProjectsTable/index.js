import React, { useRef } from 'react';
import { SearchOutlined } from '@ant-design/icons';
import PropTypes from 'prop-types';
import {
  Button,
  Input,
  Popconfirm,
  Space,
  Tag,
  Tooltip,
  Typography,
} from 'antd';

import { CommonTable } from 'components';

const ProjectsTable = ({
  loading,
  projects,
  selectedProjects,
  handleClickDelete,
  handleClickProject,
  handleSelectProjects,
  handleFetchPaginatedProjects,
}) => {
  const searchInputRef = useRef(null);

  const handleSearch = (selectedKeys, confirm) => {
    confirm();
    handleFetchPaginatedProjects(selectedKeys[0]);
  };

  const handleReset = (clearFilters) => {
    clearFilters();
    handleFetchPaginatedProjects('');
  };

  const columnsConfig = [
    {
      title: <strong>Nome do Projeto</strong>,
      dataIndex: 'name',
      key: 'name',
      width: '20%',
      filterDropdown(filterDropdownProps) {
        const { setSelectedKeys, selectedKeys, confirm, clearFilters } =
          filterDropdownProps;

        return (
          <div style={{ padding: 8 }}>
            <Input
              ref={searchInputRef}
              placeholder={`Nome`}
              value={selectedKeys[0]}
              onPressEnter={() => handleSearch(selectedKeys, confirm)}
              style={{ width: 188, marginBottom: 8, display: 'block' }}
              onChange={(e) =>
                setSelectedKeys(e.target.value ? [e.target.value] : [])
              }
            />
            <Space>
              <Button
                type='primary'
                size='small'
                style={{ width: 90 }}
                icon={<SearchOutlined />}
                onClick={() => handleSearch(selectedKeys, confirm)}
              >
                Buscar
              </Button>
              <Button
                size='small'
                style={{ width: 90 }}
                onClick={() => handleReset(clearFilters)}
              >
                Resetar
              </Button>
            </Space>
          </div>
        );
      },
      onFilterDropdownVisibleChange: (visible) => {
        if (visible) {
          setTimeout(() => searchInputRef.current.select());
        }
      },
      filterIcon(filtered) {
        return (
          <SearchOutlined style={{ color: filtered ? '#1890ff' : undefined }} />
        );
      },
      render(value, record) {
        return (
          <Button
            type='link'
            style={{ textAlign: 'left', width: '100%' }}
            onClick={() => handleClickProject(record.uuid)}
          >
            <Typography.Text
              style={{ color: '#0050B3', width: '100%' }}
              ellipsis
            >
              <Tooltip placement='topLeft' title={value}>
                {value}
              </Tooltip>
            </Typography.Text>
          </Button>
        );
      },
    },
    {
      title: <strong>Descrição</strong>,
      dataIndex: 'description',
      key: 'description',
      width: '30%',
      ellipsis: {
        showTitle: false,
      },
      render(value) {
        return (
          <Tooltip placement='topLeft' title={value}>
            {value}
          </Tooltip>
        );
      },
    },
    {
      title: <strong>Tags</strong>,
      dataIndex: 'tags',
      key: 'tags',
      width: '28%',
      filters: [
        {
          text: 'Experimentação',
          value: 'hasExperiment',
        },
        {
          text: 'Pré-implantação',
          value: 'hasPreDeployment',
        },
        {
          text: 'Implantado',
          value: 'hasDeployment',
        },
      ],
      onFilter(value, record) {
        /**
         * Utilizamos o value (hasDeployment, hasPreDeployment, hasExperiment)
         * para acessar a chave do objeto (hasDeployment, hasPreDeployment,
         * hasExperiment)
         */
        return record[value];
      },
      render(_, record) {
        return (
          <>
            {record.hasExperiment && <Tag color='purple'>Experimentação</Tag>}
            {record.hasPreDeployment && (
              <Tag color='volcano'>Pré-implantação</Tag>
            )}
            {record.hasDeployment && <Tag color='green'>Implantado</Tag>}
          </>
        );
      },
    },
    {
      title: <strong>Última modificação</strong>,
      dataIndex: 'updatedAt',
      key: 'updatedAt',
      width: '12%',
      sorter: (a, b) => {
        const dateA = new Date(a.updatedAt);
        const dateB = new Date(b.updatedAt);
        return dateA.getTime() - dateB.getTime();
      },
      sortDirections: ['descend', 'ascend'],
      render: (value) => new Date(value).toLocaleString(),
    },
    {
      title: <strong>Ação</strong>,
      dataIndex: 'action',
      key: 'action',
      width: '10%',
      render(_, record) {
        return (
          <Popconfirm
            title='Você tem certeza que deseja excluir esse projeto?'
            onConfirm={() => handleClickDelete(record.uuid)}
            okText='Sim'
            cancelText='Não'
          >
            <Button type='link'>Excluir</Button>
          </Popconfirm>
        );
      },
    },
  ];

  return (
    <CommonTable
      className='projectsTable'
      rowKey={(record) => record.uuid}
      rowSelection={{
        selectedRowKeys: selectedProjects,
        onChange: (selectedRowKeys) => {
          handleSelectProjects(selectedRowKeys);
        },
      }}
      dataSource={projects}
      columns={columnsConfig}
      pagination={false}
      isLoading={loading}
    />
  );
};

ProjectsTable.propTypes = {
  loading: PropTypes.bool.isRequired,
  projects: PropTypes.arrayOf(PropTypes.object).isRequired,
  selectedProjects: PropTypes.arrayOf(PropTypes.string).isRequired,
  handleClickDelete: PropTypes.func.isRequired,
  handleClickProject: PropTypes.func.isRequired,
  handleSelectProjects: PropTypes.func.isRequired,
  handleFetchPaginatedProjects: PropTypes.func.isRequired,
};

export default ProjectsTable;
