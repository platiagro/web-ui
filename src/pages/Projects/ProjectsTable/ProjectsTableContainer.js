import React, { useLayoutEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';

import { useIsLoading } from 'hooks';
import { showNewProjectModal } from 'store/ui/actions';
import { MyProjectsEmptyPlaceholder } from 'components/EmptyPlaceholders';
import {
  Actions as projectsActions,
  Selectors as projectsSelectors,
  PROJECTS_TYPES,
} from 'store/projects';

import ProjectsTable from './index';
import ProjectsTablePagination from '../ProjectsTablePagination/ProjectsTablePaginationContainer';

const { getProjects, getSelectedProjects, getSearchText } = projectsSelectors;

const {
  fetchPaginatedProjectsRequest,
  selectProjects,
  deleteProjectsRequest,
} = projectsActions;

const ProjectsTableContainer = () => {
  const history = useHistory();
  const dispatch = useDispatch();

  const projects = useSelector(getProjects);
  const searchText = useSelector(getSearchText);
  const selectedProjects = useSelector(getSelectedProjects);

  useLayoutEffect(() => {
    dispatch(fetchPaginatedProjectsRequest(undefined, 1, 10));
  }, [dispatch]);

  const loading = useIsLoading(
    PROJECTS_TYPES.FETCH_PROJECTS_REQUEST,
    PROJECTS_TYPES.DELETE_PROJECTS_REQUEST
  );

  const handleClickProject = (projectUuid) => {
    history.push(`/projetos/${projectUuid}`);
  };

  const handleFetchPaginatedProjects = (name) => {
    dispatch(fetchPaginatedProjectsRequest(name, 1, 10));
  };

  const handleDeleteProject = (projectUuid) => {
    dispatch(deleteProjectsRequest([projectUuid]));
  };

  const handleShowNewProjectModal = (record) => {
    dispatch(showNewProjectModal(record));
  };

  const handleSelectProjects = (record) => {
    dispatch(selectProjects(record));
  };

  return loading || searchText || (projects && projects.length > 0) ? (
    <div className='myProjectsTableContainer'>
      <ProjectsTable
        loading={loading}
        projects={projects}
        selectedProjects={selectedProjects}
        handleClickProject={handleClickProject}
        handleClickDelete={handleDeleteProject}
        handleFetchPaginatedProjects={handleFetchPaginatedProjects}
        handleShowNewProjectModal={handleShowNewProjectModal}
        handleSelectProjects={handleSelectProjects}
      />
      <br />
      <ProjectsTablePagination />
    </div>
  ) : (
    <MyProjectsEmptyPlaceholder />
  );
};

export default ProjectsTableContainer;
