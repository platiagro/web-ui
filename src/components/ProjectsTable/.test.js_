import React from 'react';

import { shallow } from 'enzyme';

import ProjectsTable from '.';

// remove after tests
const projects = [
  {
    key: '1',
    projectName: 'workshp_ForAgri_0',
    experimentsList: [
      'Experimento 0',
      'Experimento 1',
      'Experimento 2',
      'Experimento 3',
      'Experimento 4',
      'Experimento 5',
      'Experimento 6',
    ],
    created: '11/10/2019 12:59:21',
  },
  {
    key: '2',
    projectName: 'workshp_ForAgri_1',
    experimentsList: ['Experimento 0', 'Experimento 1', 'Experimento 2'],
    created: '11/10/2019 12:59:21',
  },
  {
    key: '3',
    projectName: 'workshp_ForAgri_2',
    experimentsList: ['Experimento 0'],
    created: '11/10/2019 12:59:21',
  },
  {
    key: '4',
    projectName: 'workshp_ForAgri_3',
    experimentsList: ['Experimento 0'],
    created: '11/10/2019 12:59:21',
  },
  {
    key: '5',
    projectName: 'workshp_ForAgri_4',
    experimentsList: ['Experimento 0'],
    created: '11/10/2019 12:59:21',
  },
  {
    key: '6',
    projectName: 'workshp_ForAgri_5',
    experimentsList: ['Experimento 0'],
    created: '11/10/2019 12:59:21',
  },
  {
    key: '7',
    projectName: 'workshp_ForAgri_6',
    experimentsList: ['Experimento 0'],
    created: '11/10/2019 12:59:21',
  },
  {
    key: '8',
    projectName: 'workshp_ForAgri_7',
    experimentsList: ['Experimento 0'],
    created: '11/10/2019 12:59:21',
  },
  {
    key: '9',
    projectName: 'workshp_ForAgri_8',
    experimentsList: ['Experimento 0'],
    created: '11/10/2019 12:59:21',
  },
  {
    key: '10',
    projectName: 'workshp_ForAgri_9',
    experimentsList: ['Experimento 0'],
    created: '11/10/2019 12:59:21',
  },
];

projects.forEach((project) => {
  project.experiments = project.experimentsList.join(', ');
});

const rowSelection = {
  selectedRowKeys: [],
  onChange: () => null,
};

describe('ProjectsTable component should', () => {
  it('renders without crashing', () => {
    shallow(
      <ProjectsTable
        enterProject={() => true}
        projectList={projects}
        rowSelection={rowSelection}
      />
    );
  });

  it('renders html correctly', () => {
    const projectsTableShallowed = shallow(
      <ProjectsTable
        enterProject={() => true}
        projectList={projects}
        rowSelection={rowSelection}
      />
    );

    expect(projectsTableShallowed).toMatchSnapshot();
  });
});
