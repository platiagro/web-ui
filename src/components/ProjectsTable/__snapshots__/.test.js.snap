// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsTable component should renders html correctly 1`] = `
<Table
  bordered={false}
  childrenColumnName="children"
  className=""
  columns={
    Array [
      Object {
        "dataIndex": "projectName",
        "key": "projectName",
        "title": "Nome do Projeto",
      },
      Object {
        "dataIndex": "experiments",
        "key": "experiments",
        "title": "Experimentos",
      },
      Object {
        "dataIndex": "created",
        "key": "created",
        "title": "Data de Criação",
      },
    ]
  }
  dataSource={
    Array [
      Object {
        "created": "11/10/2019 12:59:21",
        "experiments": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Experimento 0, Experimento 1, Experimento 2, Experimento 3, Experimento 4, Experimento 5, Experimento 6"
          transitionName="zoom-big-fast"
        >
          <span>
            Experimento 0, Experimento 1, Experimento 2, Experimento 3, Experimento 4, Experimento 5, Experimento 6
          </span>
        </Tooltip>,
        "experimentsList": Array [
          "Experimento 0",
          "Experimento 1",
          "Experimento 2",
          "Experimento 3",
          "Experimento 4",
          "Experimento 5",
          "Experimento 6",
        ],
        "key": "1",
        "projectName": "workshp_ForAgri_0",
      },
      Object {
        "created": "11/10/2019 12:59:21",
        "experiments": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Experimento 0, Experimento 1, Experimento 2"
          transitionName="zoom-big-fast"
        >
          <span>
            Experimento 0, Experimento 1, Experimento 2
          </span>
        </Tooltip>,
        "experimentsList": Array [
          "Experimento 0",
          "Experimento 1",
          "Experimento 2",
        ],
        "key": "2",
        "projectName": "workshp_ForAgri_1",
      },
      Object {
        "created": "11/10/2019 12:59:21",
        "experiments": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Experimento 0"
          transitionName="zoom-big-fast"
        >
          <span>
            Experimento 0
          </span>
        </Tooltip>,
        "experimentsList": Array [
          "Experimento 0",
        ],
        "key": "3",
        "projectName": "workshp_ForAgri_2",
      },
      Object {
        "created": "11/10/2019 12:59:21",
        "experiments": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Experimento 0"
          transitionName="zoom-big-fast"
        >
          <span>
            Experimento 0
          </span>
        </Tooltip>,
        "experimentsList": Array [
          "Experimento 0",
        ],
        "key": "4",
        "projectName": "workshp_ForAgri_3",
      },
      Object {
        "created": "11/10/2019 12:59:21",
        "experiments": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Experimento 0"
          transitionName="zoom-big-fast"
        >
          <span>
            Experimento 0
          </span>
        </Tooltip>,
        "experimentsList": Array [
          "Experimento 0",
        ],
        "key": "5",
        "projectName": "workshp_ForAgri_4",
      },
      Object {
        "created": "11/10/2019 12:59:21",
        "experiments": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Experimento 0"
          transitionName="zoom-big-fast"
        >
          <span>
            Experimento 0
          </span>
        </Tooltip>,
        "experimentsList": Array [
          "Experimento 0",
        ],
        "key": "6",
        "projectName": "workshp_ForAgri_5",
      },
      Object {
        "created": "11/10/2019 12:59:21",
        "experiments": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Experimento 0"
          transitionName="zoom-big-fast"
        >
          <span>
            Experimento 0
          </span>
        </Tooltip>,
        "experimentsList": Array [
          "Experimento 0",
        ],
        "key": "7",
        "projectName": "workshp_ForAgri_6",
      },
      Object {
        "created": "11/10/2019 12:59:21",
        "experiments": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Experimento 0"
          transitionName="zoom-big-fast"
        >
          <span>
            Experimento 0
          </span>
        </Tooltip>,
        "experimentsList": Array [
          "Experimento 0",
        ],
        "key": "8",
        "projectName": "workshp_ForAgri_7",
      },
      Object {
        "created": "11/10/2019 12:59:21",
        "experiments": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Experimento 0"
          transitionName="zoom-big-fast"
        >
          <span>
            Experimento 0
          </span>
        </Tooltip>,
        "experimentsList": Array [
          "Experimento 0",
        ],
        "key": "9",
        "projectName": "workshp_ForAgri_8",
      },
      Object {
        "created": "11/10/2019 12:59:21",
        "experiments": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Experimento 0"
          transitionName="zoom-big-fast"
        >
          <span>
            Experimento 0
          </span>
        </Tooltip>,
        "experimentsList": Array [
          "Experimento 0",
        ],
        "key": "10",
        "projectName": "workshp_ForAgri_9",
      },
    ]
  }
  indentSize={20}
  loading={false}
  locale={Object {}}
  onRow={[Function]}
  pagination={
    Object {
      "pageSize": 9,
    }
  }
  rowKey="key"
  rowSelection={
    Object {
      "onChange": [Function],
      "selectedRowKeys": Array [],
    }
  }
  showHeader={true}
  size="default"
  sortDirections={
    Array [
      "ascend",
      "descend",
    ]
  }
  useFixedHeader={false}
/>
`;
